docs/nachet-inference-documentation.md

```diff
@@ -41,90 +43,246 @@ end
+
+|Key parameters | Expected Value|
+|--|--|
+|model_name | The name of the pipeline|
+|folder_name | The folder where the image is uploaded in the user's container|
+|container_name | The user's container|
+|imageDims | The dimension of the image|
+|image | The image encoded in b64 (ASCII)|
+
+Note that since the information is received from the frontend, the model_name is
+an abstraction for a pipeline.
+
+The inference request will return a list with the following information:
+|key parameters | hierarchy Levels | Return Value |
+|--|--|--|
+|Filename| 0 | Contains the filename of the image|
+|Boxes | 0 | Contains all the boxes returned by the inference request|
+|labelOccurence | 0 | Contains the number of label occurence|
+|totalBoxes | 0 | Boxes total number|
+|Box | 1 | Contains all the information of one seed in the image|
+|label | 1 | Contains the top label for the seed|
+|score | 1 | Contains the top score for the seed|
+|topN | 1 | Contains the top N scores for the seed|
+|overlapping | 1 | Contains a boolean to tell if the box overlap with another one|
+|overlappingIndices | 1 | Contains the index of the overlapping box|
+|topX | 2 | The top x value of the box around a seed|
+|topY | 2 | The top y value of the box around a seed|
+|bottomX | 2 | The bottom x value of the box around a seed|
+|bottomY| 2 | The bottom y value of the box around a seed|
+
+*for more look at [nachet-model-documentation](https://github.com/ai-cfia/nachet-backend/blob/51-implementing-2-models/docs/nachet-model-documentation.md#return-value-of-models)*
+
+**topN** contains the top 5 predictions of the models:
+
+```json
+"topN": [
+    {
+        "label": "seed_name",
+        "score": 0.75
+    }
+    {
+        "label": "seed_name",
+        "score": 0.18
+    }
+    {
+        "label": "seed_name",
+        "score": 0.05
+    }
+    {
+        "label": "seed_name",
+        "score": 0.019
+    }
+    {
+        "label": "seed_name",
+        "score": 0.001
+    }
+]
+```
+
+### Blob storage and Pipeline versioning
+
+To keep track of the various pipelines iterations and versions, JSON files are
+stored in the blob storage. Users can add the JSON to the blob storage
+using the `pipelines_version_insertion.py` script. This allows for easy
+management of model and pipeline history.
+
+To use the script, 3 environment variables are necessary:
+
+* NACHET_BLOB_PIPELINE_NAME
+  * Containing the blob name where the pipelines are stored
+* NACHET_BLOB_PIPELINE_VERSION
+  * Containing the version the user wants to select
+* NACHET_BLOB_PIPELINE_DECRYPTION_KEY
+  * The key to decrypt sensible data such as the API key and the endpoint of a model.
+
+#### In the code
+
+In the backend, the pipelines are retrieved using the `get_pipelines` function.
+This function retrieved the data from the blob storage and stored the pipeline in
+the `CACHE["endpoint"]` variable. This the variable that feed the frontend the `models`
+information and metadata.
```

Comment:
This **is** the variable that feed**s** the `models`
information and metadata to the frontend.